src/main/java/org/isf/telemetry/TelemetryGUI.java

```diff
@@ -0,0 +1,239 @@
+ * Copyright © 2006-2021 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+ *
+ * Open Hospital is a free and open source software for healthcare data management.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isf.telemetry;
+
+import java.awt.AWTEvent;
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.Toolkit;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EventListener;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JDialog;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.SpringLayout;
+import javax.swing.event.EventListenerList;
+
+import org.isf.envdatacollector.AbstractDataCollector;
+import org.isf.generaldata.MessageBundle;
+import org.isf.menu.gui.MainMenu;
+import org.isf.telemetry.manager.TelemetryManager;
+import org.isf.telemetry.model.Telemetry;
+import org.isf.telemetry.util.TelemetryUtils;
+import org.isf.utils.ExceptionUtils;
+import org.isf.utils.layout.SpringUtilities;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.core.annotation.AnnotationAwareOrderComparator;
+
+public class TelemetryGUI extends JDialog {
+
+	private static final long serialVersionUID = 891561833857381224L;
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(Telemetry.class);
+	private MainMenu parent;
+	private EventListenerList telemetryListeners = new EventListenerList();
+	private Telemetry telemetry;
+
+	public TelemetryGUI(MainMenu parent) {
+		super(parent, MessageBundle.getMessage("angal.login.title"), true);
+		this.parent = parent;
+		this.addTelemetryListener(parent);
+		ApplicationContext springSexyContext = new ClassPathXmlApplicationContext("applicationContext.xml");
+		TelemetryManager telemetryManager = springSexyContext.getBean(TelemetryManager.class);
+		TelemetryUtils telemetryUtils = springSexyContext.getBean(TelemetryUtils.class);
```

Comment:
Provider is already in the code:

```suggestion
		TelemetryManager telemetryManager = Context.getApplicationContext().getBean(TelemetryManager.class);
		TelemetryUtils telemetryUtils = Context.getApplicationContext().getBean(TelemetryUtils.class);
```

Reply:
corrected